src/System.Net.Sockets/tests/FunctionalTests/SocketAPMExtensions.cs

```diff
@@ -29,12 +29,17 @@ public static void ReceiveAPM(this Socket socket, byte[] buffer, int offset, int
             socket.BeginReceive(buffer, offset, count, flags, callback, socket);
         }
 
-        public static void SendFileAPM(this Socket socket, string filename, byte[] preBuffer, byte[] postBuffer, TransmitFileOptions flags, Action handler)
+        public static void SendFileAPM(this Socket socket, string filename, byte[] preBuffer, byte[] postBuffer, TransmitFileOptions flags, Action<Exception> handler)
```

Comment:
Yeah.  I'll do so.  We're potentially eating a variety of exceptions with this the way it's written.